### ISSUE TITLE
Issue in making a subsystems.complex blast searchable db using makeblastdb

### ISSUE THREAD

**Issue by GabrieleNocchi** [transcript/subsystems#1]
Hi,

I am trying to use your discoveries and work (thanks) to make a blast searchable database of the SEED subsystems.

By only using the simplifier script on subsystem.complex the resulting FASTA gives me issues; if I use makeblastdb with the option -parse_seqids (which I need to use) it tells me that there are duplicate IDs. Tried removing duplicates, but the error persists. I think makeblastdb does not like the way these IDs are formatted. 

The error:

>BLAST Database creation error: Error: Duplicate seq_ids are found: 
>GNL|FIG|66692.3.PEG.217

P.S. Also, after using your simplifier script you also need to run this:
sed -i '1d' subsystems.complex.modified && sed -i '$ d' subsystems.complex.modified

Because there is a line at the beginning and one empty at the end, both cause problems with makeblastdb.

On the other side, the rebuilder script creates more complete IDs but they also cause issues with makeblastdb.

I would like to ask you if you actually made a blast searchable database using makeblastdb out of this, and if so, if you could quickly explain the step that you took.

**Comment by GabrieleNocchi** [comment]
Managed to sort it in a very patchy way for the moment, but lost a lot of data:

So, after using your simplifier script I get the file subsystems.complex.modified (7.1GB)

I use this command to remove first and last line:
> sed -i '1d' subsystems.complex.modified && sed -i '$ d' subsystems.complex.modified

Then this to remove duplicates (Yes there were duplicates in subsystems.complex.modified):
> sed -e '/^>/s/$/@/' -e 's/^>/#/' subsystems.complex.modified | tr -d '\n' | tr "#" "\n" | tr "@" "\t" | sort -u -k1,1 | sed -e 's/^/>/' -e 's/\t/\n/' > fileout.fa


Then:
> sed -i '1d' fileout.fa

Because the previous command added a > at the beginning of the file, so I had to remove first line again.

Then I need to remove any empty entry, because they also cause problems with makeblastdb (BLAST Database creation error: FASTA-Reader: No residues given):
> awk -v RS=">" -v FS="\n" -v ORS="" ' { if ($2) print ">"$0 } ' fileout.fa > file.fa

At the end I have this file.fa which is 4.1 GB, which can be made into a blast searchable database with:

>  makeblastdb -in file.fa -dbtype prot -out systems_db -parse_seqids

**Comment by transcript** [comment]
Hi Gabriele, 

I'm reproducing this, and I'll update with my findings.  Thanks for bringing this to my attention.

**Comment by transcript** [comment]
Hi Gabriele,

I've done some testing, and it looks like you're right, in that there are a lot of duplicates in the merged file (subsystems.complex.reduced).  I successfully made a BLAST database by doing the following steps:

1. Downloaded the appropriate files from the Subsystems FTP link
2. Ran:
  `subsys_db_rebuilder.py subsystems.complex subsystems2role subsystems2peg subsys.txt`
3. Replaced the tabs with spaces:
  `sed 's/\t/     /g' subsystems.complex.merged > notabs.subsystems.complex.merged`
4. Ran a new script to remove duplicate entries:
  `duplicate_counter.py notabs.subsystems.complex.merged`
5. Successfully made a BLAST database with the -parse_seqids option enabled:
  `makeblastdb -in notabs.subsystems.complex.merged.reduced -dbtype 'prot' -out testBlastDB -parse_seqids`

It looks like you're right, and a high number of sequences (3.3M out of 7.8M total) are duplicates.  The new script I've added, duplicate_counter.py, just reads through the tab-stripped merged subsystems file and prints only unique sequences with their annotations.

I'm not totally sure where all the duplicates came from, but I'll look into it.  In the meantime, I'll update the Readme to explain this step - it should probably be standard, as it almost halves the total final database size.
